File: Toy-Problems/Fizzbuzz.js
/**

Write a program that outputs the string representation of numbers from 1 to n.

But for multiples of three it should output “Fizz” instead of the number and for the multiples of five output “Buzz”. For numbers which are multiples of both three and five output “FizzBuzz”.


 * @param {number} n
 * @return {string[]}
 */

var fizzBuzz = function(n) {
  let results = [];
  for (let i = 1; i <= n; i++) {
    let previousResultsLength = results.length;
    if (i % 3 === 0 && i % 5 === 0) {
      results.push('FizzBuzz');
    } else if (i % 3 === 0) {
      results.push('Fizz');
    } else if (i % 5 === 0) {
      results.push('Buzz');
    }
    if (results.length === previousResultsLength) {
      results.push(String(i));
    }
  }
  return results;
};

File: Toy-Problems/Java/Fizzbuzz.java

/*

FizzBuzz

Print integers up to given value. For ints divisible by 3, print Fizz, for ints divisible by 5, print Buzz, and for ints divisible by both, print FizzBuzz.

*/

public class FizzBuzz {	
	public static void main(String[] args) {
        
		int testInt = 200;
		
		for (int i = 0; i <= testInt; i++) {
			if (i % 3 == 0 && i % 5 == 0) {
				System.out.println("FizzBuzz"); 
			} else if (i % 3 == 0) {
				System.out.println("Fizz"); 
			} else if (i % 5 == 0) {
				System.out.println("Buzz"); 
			} else {
				System.out.println(i);
			}
		}	
	}
}
